gateway/router.go

```diff
@@ -16,15 +17,22 @@ type Router struct {
 	Gateways         map[string]*Gateway
 	Message          chan config.Message
 	MattermostPlugin chan config.Message
+
+	logger *logrus.Entry
 }
 
-func NewRouter(cfg config.Config, bridgeMap map[string]bridge.Factory) (*Router, error) {
+// NewRouter initializes a new Matterbridge router for the specified configuration and
+// sets up all required gateways.
+func NewRouter(rootLogger *logrus.Logger, cfg config.Config, bridgeMap map[string]bridge.Factory) (*Router, error) {
+	logger := rootLogger.WithField("prefix", "router")
```

Comment:
```suggestion
	logger := rootLogger.WithFields(logrus.Fields{"prefix": "router"})
```

Reply:
That's really weird. That should not make a difference. 😱